lib/middleware/nunjucks-configuration/nunjucks-configuration.js

```diff
@@ -56,8 +57,8 @@ const nunjucksConfiguration = (app, components = [], options = {}) => {
     }
   }
 
-  // Reverse component path order as first matched wins
-  components.reverse().forEach(component => {
+  // Create a copy of the components array then reverse its order -- first matched wins
+  components.slice().reverse().forEach(component => {
```

Comment:
Using `slice` here simply creates an exact copy of the array so that the subsequent `reverse` invocation operates on it, and not the original array